benchmark/spmv/spmv.cpp

```diff
@@ -96,6 +101,47 @@ void apply_spmv(const char *format_name, std::shared_ptr<gko::Executor> exec,
             system_matrix->apply(lend(b), lend(x_clone));
             exec->synchronize();
         }
+
+        // tuning run
+#ifdef GINKGO_BENCHMARK_ENABLE_TUNING
+        auto &format_case = spmv_case[format_name];
+        if (!format_case.HasMember("tuning")) {
+            format_case.AddMember(
+                "tuning", rapidjson::Value(rapidjson::kObjectType), allocator);
+        }
+        auto &tuning_case = format_case["tuning"];
+        add_or_set_member(tuning_case, "time",
+                          rapidjson::Value(rapidjson::kArrayType), allocator);
+        add_or_set_member(tuning_case, "values",
+                          rapidjson::Value(rapidjson::kArrayType), allocator);
+
+        // Enable tuning for this portion of code
+        gko::_tuning_flag = true;
+        // Select some values we want to tune.
+        std::vector<gko::size_type> tuning_values{
+            1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096};
```

Comment:
could you add the output to json for these tuning value?
I am not sure whether using command parameter to adjust these setting is necessary or not. Need pass via string and handle it.

Reply:
This is done, right now it creates two arrays, like the following:
```
         "tuning": {
            "time": [
                0.0179852619,
                0.010033904300000002,
                0.0064128235,
                0.005212025800000001,
                0.0049412771000000008,
                0.0045020674,
                0.0042101267,
                0.004052650200000001,
                0.0040126375,
                0.004072723,
                0.0039625192,
                0.0042129298,
                0.0043702082000000009
            ],
            "values": [
                1,
                2,
                4,
                8,
                16,
                32,
                64,
                128,
                256,
                512,
                1024,
                2048,
                4096
            ]
        }
```

Reply:
As for passing in the values via command line, I think it's better if it stays like this for now since anyway when someone will want to tune something else, the code for this might change.

Reply:
you are right. I miss the value output. Thanks!